### ISSUE TITLE
Problem using pre-created environment with WinForms WebView2 control

### ISSUE THREAD

**Issue by ebriangast** [MicrosoftEdge/WebView2Feedback#177]
I've followed the Getting Started with WebView2 in Windows Forms documentation and am now trying to extend the application to pre-create the CoreWebView2Environment with a specific user data folder.

It seems the setting of WebView2.Source in the designer generated code triggers the automatic initialization of the environment before I have a chance to call EnsureCoreWebView2Async to provide my own. I see this happen even with the WebView2 control Source property default value of "about:blank", and if I try to remove the value for the Source property in designer I get an error of "Property value is not valid."

Specifically, I added code to Main to create the environment and pass it to the Form constructor:
 ```C#
            String userDataFolder = Path.GetTempPath() + @"NETWebView2UserDataTest";

            CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions();
            CoreWebView2Environment env = CoreWebView2Environment.CreateAsync("", userDataFolder, options).GetAwaiter().GetResult();

            Application.Run(new Form1(env));
```

Then immediately called EnsureCoreWebView2Async after calling InitializeComponent:
```C#
        public Form1(CoreWebView2Environment webView2Env = null)
        {
            InitializeComponent();
            webView.EnsureCoreWebView2Async(webView2Env);
            webView.CoreWebView2Ready += WebView_CoreWebView2Ready;

            this.Resize += new System.EventHandler(this.Form_Resize);
        }

        private void WebView_CoreWebView2Ready(object sender, EventArgs e)
        {
            webView.CoreWebView2.Navigate(addressBar.Text);
        }
```
And also set an appropriate default of addressBar.Text in the designer.

If I manually comment out the setting of webView.Source in the Designer.cs file this works, and the environment I constructed is used. But making additional changes in the designer reverts this change to the generated code.

[AB#26434822](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/26434822)

**Comment by david-risney** [comment]
Thanks for reporting this @ebriangast! Great issue. We'll look into how we might address this.

**Comment by igotbass** [comment]
@ebriangast thank you as I spent the last 2 hours wondering why this is not working.

Same issue, my use case is a application that is being delivered via RemoteApp where each user has own directory, if source is set it will not be able to set the user directory. Comment out the source solves the issue. 

Thanks

**Comment by champnic** [comment]
This should be fixed now - setting Source to about:blank should avoid the implicit initialization and this error. If you still see this please reopen. Thanks!

**Comment by champnic** [comment]
Apologies, apparently we have a fix but it hasn't been committed yet. This should be fixed soon!

**Comment by wuhi** [comment]
any news on this?
i'm not able to use a precreated environment with webview2. so i cannot set the userDataFolder

**Comment by champnic** [comment]
We're making a change to the default behavior here so that the Source property doesn't get automatically set for Winforms. Instead the default is null, and the dev can choose to set Source if they want implicit initialization, or can manually create the environment and initialize explicitly. Additionally, we are adding a [CoreWebView2CreationProperties](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/wpf/0-9-515/microsoft-web-webview2-wpf-corewebview2creationproperties) like the existing WPF one, to allow easily setting the environment while still using the Source property. These changes will be coming in about 2 weeks. Thanks!

**Comment by champnic** [comment]
This should be fixed in [WebView2 SDK 1.0.674-prerelease](https://www.nuget.org/packages/Microsoft.Web.WebView2/1.0.674-prerelease). If this is still an issue please let us know. Thanks!
